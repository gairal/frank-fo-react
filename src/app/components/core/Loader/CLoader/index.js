import CLoader from './CLoader';

export default CLoader;
