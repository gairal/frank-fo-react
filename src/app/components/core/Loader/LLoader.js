import React from 'react';
import { LinearProgress } from '@material-ui/core';

export default () => <LinearProgress mode="query" />;
