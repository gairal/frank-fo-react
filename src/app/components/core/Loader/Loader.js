import React from 'react';
import PropTypes from 'prop-types';
import { LinearProgress } from 'material-ui/Progress';
import { Grid } from 'material-ui';

export const Loader = props => (
  props.isFetching
    ? (
      <Grid
        container
        alignItems="center"
        direction="row"
        justify="center"
        style={{ width: '100%', display: 'block' }}
      >
        <LinearProgress mode="query" />
      </Grid>
    )
    : null
);

Loader.propTypes = {
  isFetching: PropTypes.bool,
};

Loader.defaultProps = {
  isFetching: false,
};

export default Loader;
