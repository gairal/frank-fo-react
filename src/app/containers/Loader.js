import { connect } from 'react-redux';
import { injectReducer } from '@/store/reducers';
import Loader from '@/components/core/Loader';

const LOADER_SHOW = 'LOADER_SHOW';
const LOADER_HIDE = 'LOADER_HIDE';

// ------------------------------------
// Action Handlers
// ------------------------------------
const ACTION_HANDLERS = {
  [LOADER_SHOW]: state => ({
    ...state,
    isFetching: true,
  }),
  [LOADER_HIDE]: state => ({
    ...state,
    isFetching: false,
  }),
};

const initialState = {
  title: '',
  mobileOpen: false,
  isFetching: false,
};

const reducer = (state = initialState, action) => {
  const handler = ACTION_HANDLERS[action.type];
  return handler ? handler(state, action) : state;
};

export default (store, type) => {
  injectReducer(store, { key: 'loader', reducer });

  const mapStateToProps = state => ({
    isFetching: state.loader.isFetching,
    type,
  });

  return connect(mapStateToProps)(Loader);
};
