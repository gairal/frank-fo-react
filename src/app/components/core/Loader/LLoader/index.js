import LLoader from './LLoader';

export default LLoader;
