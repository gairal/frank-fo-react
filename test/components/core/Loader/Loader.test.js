import React from 'react';
import renderer from 'react-test-renderer';
import Loader from '@/components/core/Loader';

describe('components > core > Loader', () => {
  it('Renders correctly default', () => {
    const component = renderer.create(<Loader />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Appears when request fetching', () => {
    const component = renderer.create(<Loader isFetching />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly linear', () => {
    const component = renderer.create(<Loader type="linear" isFetching />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders correctly circular', () => {
    const component = renderer.create(<Loader type="circular" isFetching />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
