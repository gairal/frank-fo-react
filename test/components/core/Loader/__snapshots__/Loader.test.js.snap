// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > core > Loader Appears when request fetching 1`] = `
<div
  className="MuiLinearProgress-root-1 MuiLinearProgress-colorPrimary-2"
  mode="query"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar-9 MuiLinearProgress-barColorPrimary-10 MuiLinearProgress-bar1Indeterminate-12"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar-9 MuiLinearProgress-barColorPrimary-10 MuiLinearProgress-bar2Indeterminate-15"
    style={Object {}}
  />
</div>
`;

exports[`components > core > Loader Renders correctly circular 1`] = `
<div
  style={
    Object {
      "marginTop": "30vh",
      "textAlign": "center",
    }
  }
>
  <div
    className="MuiCircularProgress-root-18 MuiCircularProgress-colorPrimary-21 MuiCircularProgress-indeterminate-20"
    role="progressbar"
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg-23"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle-24 MuiCircularProgress-circleIndeterminate-26"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`components > core > Loader Renders correctly default 1`] = `null`;

exports[`components > core > Loader Renders correctly linear 1`] = `
<div
  className="MuiLinearProgress-root-1 MuiLinearProgress-colorPrimary-2"
  mode="query"
  role="progressbar"
>
  <div
    className="MuiLinearProgress-bar-9 MuiLinearProgress-barColorPrimary-10 MuiLinearProgress-bar1Indeterminate-12"
    style={Object {}}
  />
  <div
    className="MuiLinearProgress-bar-9 MuiLinearProgress-barColorPrimary-10 MuiLinearProgress-bar2Indeterminate-15"
    style={Object {}}
  />
</div>
`;
